### PR TITLE
Add global-job-replicas label/annotation

### LINKED ISSUE #676: Add global replica count label/annotation to support multislice training workloads with different templates
**What would you like to be added**:

Add a label and annotation `jobset.sigs.k8s.io/global-job-replicas` which will contain the total number of replicas in the JobSet.

**Why is this needed**:

Some users want to run multislice training workloads using replicated jobs with different templates. #617 and #649 added important features to make this possible, but the use case is still not supported because the workloads need to know the total number of replicas. The new label and annotation will finally address this.

### PR DESCRIPTION
**Pull request by GiuseppeTT**
/kind feature

Fixes #676

### CODE REVIEW

**Comment by googs1025** [comment]
/ok-to-test
@GiuseppeTT 
can you fix the EasyCLA? :)

**Comment by GiuseppeTT** [comment]
> /ok-to-test @GiuseppeTT can you fix the EasyCLA? :)

Done : )

**Comment by danielvegamyhre** [comment]
/lgtm

@GiuseppeTT is this going to be used to set TPU_NUM_SLICES (or similar) env var?

File: api/jobset/v1alpha2/jobset_types.go [hunk 1]
```diff
@@ -22,6 +22,10 @@ import (
 const (
 	JobSetNameKey         string = "jobset.sigs.k8s.io/jobset-name"
 	ReplicatedJobReplicas string = "jobset.sigs.k8s.io/replicatedjob-replicas"
+	// GlobalJobReplicasKey is a label/annotation set to the total number of replicatedJob replicas.
+	// For each JobSet, this value will be equal to the sum of `replicas`, where `replicas`
+	// is equal to jobset.spec.replicatedJobs[*].replicas.
+	GlobalJobReplicasKey string = "jobset.sigs.k8s.io/global-job-replicas"
```
Comment (ahg-g) [inline api/jobset/v1alpha2/jobset_types.go:28]: `jobset.sigs.k8s.io/global-job-count`

Comment (ahg-g) [reply api/jobset/v1alpha2/jobset_types.go:28]: Actually, we used `jobset.sigs.k8s.io/replicatedjob-replicas` above, so the equivalent here would be `jobset.sigs.k8s.io/global-replicas` perhaps.

**Comment by GiuseppeTT** [comment]
> @GiuseppeTT is this going to be used to set TPU_NUM_SLICES (or similar) env var?

Yes! It is going to be used to set the env variable `MEGASCALE_NUM_SLICES`.

File: api/jobset/v1alpha2/jobset_types.go (hunk 1 above)
Comment (GiuseppeTT) [reply api/jobset/v1alpha2/jobset_types.go:28]: Hi Abdullah.

Just to be extra sure. Are you suggesting to rename `jobset.sigs.k8s.io/global-job-replicas` to `jobset.sigs.k8s.io/global-replicas`? If so, I agree, it's a better name and I can change it : )

Comment (ahg-g) [reply api/jobset/v1alpha2/jobset_types.go:28]: Yes, to align with the other annotation we have for number of replicas within a replicatedJob

Comment (GiuseppeTT) [reply api/jobset/v1alpha2/jobset_types.go]: Done!

**Pull request merged**